src/pages/index.js

```diff
@@ -73,31 +105,96 @@ function handleImageClick(cardData) {
   imageModal.open(cardData);
 }
 
+function handleLikeClick(card) {
+  api
+    .likeCard(card._id, card._isLiked)
+    .then(() => {
+      card.handleLikeClick();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+
 function handleProfileModalSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.subtext);
-  //   profileTitle.textContent = inputValues.title;
-  //   profileSubtext.textContent = inputValues.subtext;
+  profileModal.setloading(true, "Saving...");
+  api
+    .setUserInfo(inputValues)
+    .then((data) => {
+      userInfo.setUserInfo(data);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileModal.setloading(false, "Save");
+    });
+
+  userInfo.getUserInfo(inputValues.name, inputValues.about);
 }
 
-function handleNewCardModalSubmit(inputValues) {
-  debugger;
-  renderCard({ name: inputValues.title, link: inputValues.link });
+function handleNewCardModalSubmit(card) {
+  cardModal.setloading(true, "Saving...");
+  api
+    .addCard({ name: card.title, link: card.link })
+    .then((data) => {
+      renderCard(data);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      cardModal.setloading(false, "Create");
+    });
   newCardModalForm.reset();
 }
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  jobSelector: ".profile__subtext",
-});
+function handleDeleteClick(card) {
+  confirmModal.open();
+  confirmModal.setSubmitHandler(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.delete();
+        confirmModal.close();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  });
+}
+
+function handleAvatarModalSubmit(url) {
+  avatarModal.setloading(true, "Saving...");
+  api
+    .setAvatar(url)
+    .then((data) => {
+      userInfo.setAvatar(data.avatar);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarModal.setloading(false, "Save");
+    });
+  avatarModalForm.reset();
+  avatarModal.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well